### ISSUE TITLE
Proposed enhancement: Search for groups

### ISSUE THREAD

**Issue by AEgit** [JabRef/jabref#1904]
I propose to add a search functionality just for groups (unfortunately my Java knowledge is nearly non-existent, so I can only propose this, but not contribute). For large databases with many groups (I have >900 groups) it is a bit of a hassle having to scroll to the group of interest. It would be nice to have a search feature (similar to what is already available articles) just for groups. When searching for a group you could then immediately hop to the group of interest, instead of having to scroll to it.

**Comment by Siedlerchr** [comment]
@oscargus  Relates to #1876, Is there a "group" pseudo field already?

**Comment by oscargus** [comment]
One can search for groups=xxx in the normal search and that will highlight/filter the entries in that group.

@AEgit Is that enough or do you mean having a search box in the groups side panel to easily filter out groups there? (Which I think is a quite reasonable way to implement it.)

**Comment by AEgit** [comment]
@oscargus 

I was thinking about a search box in the groups sides panel - the issue with filtering is, that you still have to scroll in the group side panel, if you want to see the actual group (not just the articles entries associated with the group). Reason for this could be, that you want to assign the group to another group (as a subgroup) - currently that can only be done by scrolling to the actual group.

Furthermore, by filtering you are not able to see entries, that are part of a subgroup of the group that you are searching for.

So, if it is possible, I would really appreciate to see an additional search box in groups side panel to filter out groups.

**Comment by tobiasdiez** [comment]
Good idea!
I think, I'll try to implement it during the redesign of the groups interface (work-in-progress, https://github.com/JabRef/jabref/pull/1438).

**Comment by AEgit** [comment]
Now that version 3.7 has been released, could we reopen this please? I would still be very interested in this functionality, and, as far as I can tell, the redesign of the groups interface (https://github.com/JabRef/jabref/pull/1438) is currently on hold, so there might be other options to implement this?

**Comment by matthiasgeiger** [comment]
This is quite some effort. And as the new interface will eventually replace the current one, we won't implement it for now as we would have to implement it twice, which is not really reasonable with our limited resources.

**Comment by AEgit** [comment]
Hmmm, ok - could we then reopen "Groups JavaFX Rework" thread (https://github.com/JabRef/jabref/pull/1438)? I just want to make sure that this problem is not forgotten. Or do people remember this issue, even when it's closed?

**Comment by tobiasdiez** [comment]
We remember everything! 😄 
Just kidding. Of course we sometimes forget open issues and get a reminder - it also helps motivation to work on these things. 
So let me give you an idea about the timeline for this issue:
- First we have to switch to the new ui-technology (called "JavaFX"). Progress is tracked at #2290. Once this is done, we reopen every issue tagged with "after-migration-to-javafx".
- Then we will implement the rest of the groups interface and so finish #1438.
- Finally, we have can implement new features as search for groups etc.

**Comment by AEgit** [comment]
Thank you very much! All right, so I'll just follow https://github.com/JabRef/jabref/pull/2290 and be excited about every progress made there :D

Would it help, if I were to install the JavaFX branch and report some bugs I would encounter? Or would this just clog up the issues reported here for the Master branch?

**Comment by koppor** [comment]
I think, currently, the most help is to point people to https://donations.jabref.org. Then we will hopefully have money to pay someone to implement that. Seeing the 180 open issues at https://github.com/JabRef/jabref/issues, the 121 open issues at https://github.com/koppor/jabref/issues

We can also put [Supporter](https://supporterhq.com/) in place to encourage continuous finding of a supporting workforce. This only makes sense, if 10 or more people donate regularly.

[![Support]( https://supporterhq.com/api/b/8rzilokn64agwryeus1y17b0r/JabRef)]( https://supporterhq.com/support/8rzilokn64agwryeus1y17b0r/JabRef )

**Comment by tobiasdiez** [comment]
This is implemented in the latest development version. Could you please check the build from http://builds.jabref.org/master/. Thanks!

**Comment by AEgit** [comment]
The search feature does appear in the groups panel. The performance, however, is currently to low for my large database to assess whether it works or not: When I insert one character, JabRef uses 100% of the CPU power and the RAM consumption is also growing to 6 GB (after which I killed the process, so maybe even more), but nothing is found is within a couple of minutes. I guess this related to the problems mentioned here: https://github.com/JabRef/jabref/issues/2599

Once these issues are solved, the new search feature will be a great help for my work! Thank you very much!

**Comment by tobiasdiez** [comment]
Thanks for the feedback! I can reproduce the really bad performance for you db. I will try to improve the performance after everything works. Lets keep this issue closed and #2599 tracks the performance issue.

**Comment by AEgit** [comment]
Thank you very much for your help!
